### PR TITLE
Feat: IP filtering middleware

### PR DESCRIPTION
**Pull request by valcker**

### CODE REVIEW

File: packages/gateway/src/middleware/mw_ip_filtering.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+const getMiddleware: RequestMiddleware = function (apiDef: ApiDef): RequestHandler {
+  // skipping if disabled
+  if (apiDef.enable_ip_filtering !== true) {
+    return (req, res, next) => next();
+  }
+
+  let ips: string[] = [];
+  let strategy: Strategy = 'none';
+  const logger = prefixLogger('mw_ip_blacklist');
```
Comment (germankrause) [inline packages/gateway/src/middleware/mw_ip_filtering.ts:18]: please move prefixed logger out of middleware, so it's called only once

File: packages/gateway/src/middleware/mw_ip_filtering.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const ipValidator = function (ip: string): boolean {
+    if (!isIP(ip)) {
+      logger.warn('Incorrect IP: "%s". Skipping...', ip);
+      return false;
+    }
+    return true;
+  };
+
+  if (apiDef.allow_ips && apiDef.allow_ips.length > 0) {
```
Comment (germankrause) [inline packages/gateway/src/middleware/mw_ip_filtering.ts:27]: apiDef?.allow_ips?.length

File: packages/gateway/src/middleware/mw_ip_filtering.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    strategy = 'allow';
+    ips = apiDef.allow_ips.filter(ipValidator);
+    logger.info('Enabling requests from the following IPs: %s', ips);
+  }
+
+  if (apiDef.deny_ips && apiDef.deny_ips.length > 0 && strategy != 'allow') {
```
Comment (germankrause) [inline packages/gateway/src/middleware/mw_ip_filtering.ts:33]: the same

Comment (germankrause) [reply packages/gateway/src/middleware/mw_ip_filtering.ts:33]: strategy != 'allow', so it cannot work as a white and black list at the same time?

**Review by germankrause — CHANGES_REQUESTED** [review]

File: packages/gateway/src/middleware/mw_ip_filtering.ts (hunk 3 above)
Comment (valcker) [reply packages/gateway/src/middleware/mw_ip_filtering.ts]: > strategy != 'allow', so it cannot work as a white and black list at the same time?

exactly

**Review by germankrause — APPROVED** [review]

**Pull request merged**